vendor/blockchain_contracts/src/bin/calculate_offsets/ethereum/rfc003/contract.rs

```diff
@@ -25,38 +32,90 @@ impl Contract {
         })
     }
 
+    fn replace_contract_offset_parameters_in_header(
+        header: &mut [u8],
+        body: &[u8],
+    ) -> Result<(), Error> {
+        let body_length = body.len();
+        let header_length = header.len();
+
+        Self::replace_offset_parameter_in_header(
+            "1001",
+            "start of contract when loading into memory",
+            header_length,
+            header,
+        )?;
+        Self::replace_offset_parameter_in_header(
+            "2002",
+            "end of contract when loading into memory",
+            body_length,
+            header,
+        )?;
+        Self::replace_offset_parameter_in_header(
+            "3003",
+            "length of contract when returning for execution",
+            body_length,
+            header,
+        )?;
+
+        Ok(())
+    }
+
+    fn replace_offset_parameter_in_header(
+        replace_pattern: &str,
+        name: &str,
```

Comment:
🔧 Not required but handy to know: 

Since you call `into()` on your parameters later on, you could make use of the power of rust and change the parameter type to a generic type which implements `Into<String>`. 

https://doc.rust-lang.org/std/convert/trait.Into.html

e.g. 
```
 fn replace_offset_parameter_in_header<T: Into<String>>(
        replace_pattern: T,
        name: T,
        value: usize,
        header: &mut [u8],
    ) -> Result<(), Error> 
```